justfile

```diff
@@ -0,0 +1,379 @@
+    @echo "  version-major    - Bump major version"
+    @echo "  version-minor    - Bump minor version"
+    @echo "  version-patch    - Bump patch version"
+
+# Run recipe
+run:
+    #!/usr/bin/env bash
+    set -euo pipefail
+    echo "Running {{APP_NAME}} v{{VERSION}} locally..."
+    just build
+    ./{{BUILD_DIR}}/{{APP_NAME}} {{RUN_ARGS}}
+
+# Build recipe
+build:
+    #!/usr/bin/env bash
+    set -euo pipefail
+    echo "Building {{APP_NAME}} v{{VERSION}}..."
+    mkdir -p {{BUILD_DIR}}
+    just gen-proto
+    just gen-templ
+    go build -ldflags="-s -w \
+        -X '{{MODULE}}/internal/utils.AppVersion={{VERSION}}' \
+        -X '{{MODULE}}/internal/utils.BuildTime={{BUILD_TIME}}' \
+        -X '{{MODULE}}/internal/utils.GitCommit={{COMMIT_HASH}}'" \
+        -trimpath -o {{BUILD_DIR}}/{{APP_NAME}} cmd/nutmix/*.go
+
+# Dependencies recipe
+deps:
+    #!/usr/bin/env bash
+    set -euo pipefail
+    echo "Checking dependencies..."
+
+    # Install Go tools (pinned versions)
+    go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.36.9
+    go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.5.1
+    go install github.com/a-h/templ/cmd/templ@v0.3.943
+
+    # Check protobuf-compiler
+    if ! command -v protoc >/dev/null 2>&1; then
+      echo "Installing protobuf-compiler..."
+      if command -v apt-get >/dev/null 2>&1; then
+        sudo apt-get update && sudo apt-get install -y protobuf-compiler
+      elif command -v brew >/dev/null 2>&1; then
+        brew install protobuf
+      else
+        echo "Please install protobuf-compiler manually for your system"
+        exit 1
+      fi
+    else
+      echo "protobuf-compiler already installed"
+    fi
+
+    # Download Go module dependencies only if needed
+    if [ ! -f go.sum ] || [ go.mod -nt go.sum ]; then
+      echo "Running go mod download..."
+      go mod download
+    else
+      echo "Go modules up to date"
+    fi
+
+    echo "Dependencies check completed"
+
+# Generate protobuf code
+gen-proto:
+    #!/usr/bin/env bash
+    set -euo pipefail
+    echo "Generating protobuf code..."
+    protoc --go_out=. --go_opt=paths=source_relative --go-grpc_out=. --go-grpc_opt=paths=source_relative --experimental_allow_proto3_optional internal/gen/signer.proto
+
+# Generate Go code from templ files
+gen-templ:
+    #!/usr/bin/env bash
+    set -euo pipefail
+    echo "Generating Go code from templ files..."
+    templ generate .
+
+
+# Dev recipe
+dev:
+    #!/usr/bin/env bash
+    set -euo pipefail
+    echo "Starting development environment..."
+    just docker-db
+    just run
+
+# Generate test keys
+gen-test-keys:
+    #!/usr/bin/env bash
+    set -euo pipefail
+    echo "MINT_PRIVATE_KEY=$(openssl rand -hex 32)"
+
+# Test recipe
+test:
+    #!/usr/bin/env bash
+    set -euo pipefail
+    echo "Running tests..."
+    go test -v ./...
+
+# Lint recipe
+lint:
+    #!/usr/bin/env bash
+    set -euo pipefail
+    echo "Running linter..."
+    golangci-lint run
```

Comment:
this command line tool should prabably be installed in the install dependencies command